src/pretix/base/models/orders.py

```diff
@@ -2659,6 +2659,20 @@ def tax_value(self):
                             self.event.currency)
         return self.price - net
 
+    @cached_property
+    def sort_key(self):
+        subevent_key = (self.subevent.date_from, str(self.subevent.name), self.subevent_id) if self.subevent_id else (0, "", 0)
+        category_key = (self.item.category.position, self.item.category.id) if self.item.category_id is not None else (0, 0)
+        item_key = self.item.position, self.item_id
+        variation_key = (self.variation.position, self.variation.id) if self.variation_id is not None else (0, 0)
+        line_key = (self.price, (self.voucher_id or 0), (self.seat.sorting_rank if self.seat_id else None), self.pk)
+        sort_key = subevent_key + category_key + item_key + variation_key + line_key
+
+        if self.addon_to_id:
+            return self.addon_to.sort_key + (1 if self.is_bundled else 2,) + sort_key
+        else:
+            return sort_key + (0,) + sort_key
```

Comment:
Is that even needed or would just `sortkey` be enough in this case? Because the sort_keys are anyways not the same length as `self.addon_to.sort_key + (1 if self.is_bundled else 2,) + sort_key` produces a longer sort_key than `sort_key + (0,) + sort_key`.

Reply:
Uh, I did not think correctly there, yes. I was intending for them to always be the same length, but in fact python deals fine with comparing different-length tuples.